onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -8118,6 +8118,128 @@ def aten_std_mean_correction(
     return op.Sqrt(var), mean
 
 
+def _center_window_around_zeros_if_needed(
+    window: TFloat, n_fft: int
+) -> TFloat:
+    # first dimension
+    n_win = op.Shape(window, start=0, end=1)
+
+    left = op.Div(op.Sub(n_fft, n_win), op.Constant(value_ints=[2]))
+
+    right = op.Sub(op.Sub(n_fft, left), n_win)
+    left = op.Reshape(left, op.Constant(value_ints=[1]))
+    right = op.Reshape(right, op.Constant(value_ints=[1]))
+
+    left_win = op.Expand(op.Constant(value_ints=[0]), left)
+    right_win = op.Expand(op.Constant(value_ints=[0]), right)
+    right_win = op.CastLike(right_win, window)
+    left_win = op.CastLike(left_win, window)
+    window_padded = op.Concat(left_win, window, right_win, axis=0)
+
+    # Center window around zeros if needed (required by ONNX's STFT)
+    window = op.Where(op.Less(op.Squeeze(n_win), n_fft), window_padded, window)
+    return window
+
+
+def _create_window_from_win_length(win_length: int, n_fft: int) -> TFloat:
+    left = op.Div(op.Sub(n_fft, win_length), op.Constant(value_ints=[2]))
+
+    right = op.Sub(op.Sub(n_fft, left), win_length)
+    left = op.Reshape(left, op.Constant(value_ints=[1]))
+    right = op.Reshape(right, op.Constant(value_ints=[1]))
```

Comment:
These op.Reshape is turning ints to be tensors, so actually line 8145 and 8147, you can still do non-op calculations (we don't need to trace those)

```python
    left = (n_fft - win_length) / 2

    right = n_fft - left - win_length
```